### PR TITLE
[GSK-2352] submit result to leaderboard ds

### PR DESCRIPTION
**Pull request by cy-moi**
Features:
- Log in with given token or env vars
- Submit cicd results to leaderboard dataset on huggingface
- Connected with leaderboard interface in huggingface space

### CODE REVIEW

File: cli.py [hunk 1]
```diff
@@ -7,11 +7,11 @@
 from giskard_cicd.loaders import GithubLoader, HuggingFaceLoader
 from giskard_cicd.pipeline.runner import PipelineRunner
 
-from automation import create_discussion
+from automation import create_discussion, init_commit_scheduler, commit_to_dataset
 
 
 logger = logging.getLogger(__file__)
-
+scheduler = init_commit_scheduler()
```
Comment (Inokinoki) [inline cli.py:14]: In case of not login (`ValueError` or some other exceptions), does the program stop?

File: automation/utils.py [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+def check_env_vars_and_login(hf_token=None, dataset_id=None):
+
+    if HF_WRITE_TOKEN is None and hf_token is None:
+        raise ValueError('HF_WRITE_TOKEN is not set')
+
+    # check the leaderboard dataset exists
+    if DATASET_ID is None and dataset_id is None:
+        raise ValueError('DATASET_ID is not set')
+    
+    login(token=HF_WRITE_TOKEN or hf_token, write_permission=True)
```
Comment (Inokinoki) [inline automation/utils.py:16]: What are the exceptions if without write perm or token invalid?

File: automation/post_discussion.py [hunk 3]
```diff
@@ -12,7 +12,7 @@ def construct_opening(dataset_id, dataset_config, dataset_split, vulnerability_c
     """
     if dataset_id is not None:  
         opening += f"""
-    \nThis automated analysis evaluated the model on the dataset {dataset_id} (subset `{dataset_config}`, split `{dataset_split}`).
+        \nThis automated analysis evaluated the model on the dataset {dataset_id} (subset `{dataset_config}`, split `{dataset_split}`).
```
Comment (Inokinoki) [inline automation/post_discussion.py:15]: See https://github.com/Giskard-AI/cicd/pull/19#discussion_r1437755269

File: cli.py (hunk 1 above)
Comment (cy-moi) [reply cli.py]: I have updated the code here to avoid stop the process at all if token or leaderboard dataset missing.

**Review by Inokinoki — APPROVED** [review]
Since exit handler `atexit.register(self._push_to_hub)` exists at https://github.com/huggingface/huggingface_hub/blob/v0.20.1/src/huggingface_hub/_commit_scheduler.py#L136C9-L136C43
It should be safe if we upload the result to the dataset in the scanner

File: cli.py [hunk 4]
```diff
@@ -124,9 +122,11 @@
         rendered_report = report.to_html()
 
     if args.output_portal == "huggingface":
+        scheduler = init_dataset_commit_scheduler(hf_token=args.hf_token, dataset_id="ZeroCommand/test-giskard-report")
```
Comment (Inokinoki) [inline cli.py:125]: Change the `dataset_id` later

**Pull request merged**